server/CGameHandler.cpp

```diff
@@ -7073,22 +7081,57 @@ void CGameHandler::handleCheatCode(std::string & cheat, PlayerColor player, cons
 			}
 		}
 	}
-	else if (cheat == "vcmiainur" || cheat == "vcmiangband" || cheat == "vcmiglaurung")
+	else if (cheat == "vcmiainur" || cheat == "vcmiangband" || cheat == "vcmiglaurung" || cheat == "vcmiarchangel"
+		|| cheat == "vcmiblackknight" || cheat == "vcmicrystal" || cheat == "vcmiazure" || cheat == "vcmifaerie")
 	{
 		cheated = true;
 		if (!hero) return;
 		///Gives N creatures into each slot
-		std::map<std::string, std::pair<int, int>> creatures;
-		creatures.insert(std::make_pair("vcmiainur", std::make_pair(13, 5))); //5 archangels
-		creatures.insert(std::make_pair("vcmiangband", std::make_pair(66, 10))); //10 black knights
-		creatures.insert(std::make_pair("vcmiglaurung", std::make_pair(133, 5000))); //5000 crystal dragons
-
-		const CCreature * creature = VLC->creh->objects.at(creatures[cheat].first);
+		std::map<std::string, std::pair<std::string, int>> creatures;
+		creatures.insert(std::make_pair("vcmiainur", std::make_pair("archangel", 5))); //5 archangels
+		creatures.insert(std::make_pair("vcmiangband", std::make_pair("blackKnight", 10))); //10 black knights
+		creatures.insert(std::make_pair("vcmiglaurung", std::make_pair("crystalDragon", 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiarchangel", std::make_pair("archangel", 5))); //5 archangels
+		creatures.insert(std::make_pair("vcmiblackknight", std::make_pair("blackKnight", 10))); //10 black knights
+		creatures.insert(std::make_pair("vcmicrystal", std::make_pair("crystalDragon", 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiazure", std::make_pair("azureDragon", 5000))); //5000 azure dragons
+		creatures.insert(std::make_pair("vcmifaerie", std::make_pair("fairieDragon", 5000))); //5000 faerie dragons
+
+		const int32_t creatureIdentifier = VLC->modh->identifiers.getIdentifier(CModHandler::scopeGame(), "creature", creatures[cheat].first, false).get();
+		const CCreature * creature = VLC->creh->objects.at(creatureIdentifier);
 		for (int i = 0; i < GameConstants::ARMY_SIZE; i++)
 			if (!hero->hasStackAtSlot(SlotID(i)))
 				insertNewStack(StackLocation(hero, SlotID(i)), creature, creatures[cheat].second);
 	}
-	else if (cheat == "vcminoldor")
+	else if (boost::starts_with(cheat, "vcmiarmy"))
+	{
+		cheated = true;
+		if (!hero) return;
+
+		std::vector<std::string> words;
+		boost::split(words, cheat, boost::is_any_of(" "));
+
+		if(words.size() < 2)
+			return;
+
+		std::string creatureIdentifier = words[1];
+
+		auto creature = vstd::tryFindIf(VLC->creh->objects,
```

Comment:
Use getIdentifier here as well

Reply:
I made that one allow case insensitive name of creatures on purpose, so you do not need to think if the creature is crystalDragon or crystaldragon etc.

Reply:
It is always crystalDragon. All H3 creatures use such format.
Also, I'd like to eventually remove direct access to (handler)->objects, so such code should be avoided

Reply:
i intended this to be used with mods, i guess i can go your way and add case-insensitive search later if needed